src/Microsoft.AspNetCore.OData/Routing/ODataActionSelector.cs

```diff
@@ -80,10 +81,23 @@ public ActionDescriptor SelectBestCandidate(RouteContext context, IReadOnlyList<
         {
             RouteData routeData = context.RouteData;
             ODataPath odataPath = context.HttpContext.ODataFeature().Path;
+
+            var routers = routeData.Routers;
+            var odataRoute = routeData.Routers.FirstOrDefault((router) =>
+            {
+                var _odataRoute = router as ODataRoute;
+                return _odataRoute == null ?
+                    false : true;
+            }) as ODataRoute;
+
+            var routePrefix = odataRoute == null ? null : odataRoute.RoutePrefix;
+            
             if (odataPath != null && routeData.Values.ContainsKey(ODataRouteConstants.Action))
             {
                 // Get the available parameter names from the route data. Ignore case of key names.
-                IList<string> availableKeys = routeData.Values.Keys.Select(k => k.ToLowerInvariant()).AsList();
+                IList<string> availableKeys = routeData.Values.Keys.Select(k => k.ToLowerInvariant())
+                    .Where((key) => routePrefix != "{" + key + "}")
```

Comment:
>(key) => routePrefix != "{" + key + "}") [](start = 27, length = 40)

what's the meaning to compare the {key} with the RoutePrefix ?

Reply:
I have found that when the Web API OData is configured with a parameterized prefix, that prefix is included in the available keys. So in that statement I was removing the prefix from the available keys in order to easily detect when a candidate action has no parameters to match the actual keys in the path. But this does feel like a hack and there's probably a better or clearer way to achieve the goal. Or maybe I just misunderstand the use case of a parameterized route prefix. Why is the route prefix added to the route data, is it meant to be bound to action parameters?